.github/scripts/sync_labels.py

```diff
@@ -0,0 +1,114 @@
+    parser.add_argument('--label_action', type=str, choices=['opened', 'labeled', 'unlabeled'], required=True, help='Sync labels action')
+    return parser.parse_args()
+
+
+def get_pr_request_body(repo, number):
+    url = f"https://api.github.com/repos/{repo}/pulls/{number}"
+    headers = {
+        'Authorization': f'token {github_token}',
+        'Accept': 'application/vnd.github.v3+json'
+    }
+    response = requests.get(url, headers=headers)
+    if response.ok:
+        pr_details = response.json()
+        return pr_details['body']
+    else:
+        print("Failed to fetch PR details. Status code:", response.status_code)
+
+
+def get_linked_issues(pr_body):
+    pattern = re.compile(r'Fixes:? (?:#|https.*?/issues/)(\d+)', re.IGNORECASE)
+    matches = re.findall(pattern, pr_body)
+    if not matches:
+        raise RuntimeError("No regex matches found in the body!")
+    issue_numbers = []
+    for match in matches:
+        issue_numbers.append(match)
+        print(f"Found issue number: {match}")
+    return issue_numbers
+
+
+def get_linked_prs(repo, issue_number, token):
+    query = f"repo:{repo} is:pr is:open linked:issue {issue_number}"
+    url = f"https://api.github.com/search/issues?q={query}"
+    headers = {"Authorization": f"Bearer {token}"}
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        prs = response.json().get('items', [])
+        return prs
+    else:
+        raise Exception(f"GitHub API returned {response.status_code}: {response.text}")
+
+
+def sync_labels(issue_numbers, number, repo, label_action):
+    print('Sync labels from Issue to linked pull request')
+    github = Github(github_token)
+    for issue_number in issue_numbers:
+        issue_labels = [label.name for label in github.get_repo(repo).get_issue(int(issue_number)).get_labels()]
+        print(issue_labels)
+        if label_action == 'opened':
+            pull_request_labels = [label.name for label in github.get_repo(repo).get_pull(int(number)).get_labels()]
+            for label in issue_labels:
+                if label and label not in pull_request_labels:
+                    github.get_repo(repo).get_pull(int(number)).add_to_labels(label)
+                    print(f"Found and added issue label: {label}")
+        else:
+            pull_request_labels = [label.name for label in github.get_repo(repo).get_pull(int(number)).get_labels() if label.name.startswith('backport/')]
+            issue_backport_label = [label.name for label in github.get_repo(repo).get_issue(int(issue_number)).get_labels() if label.name.startswith('backport/')]
+            filter_pull_request_labels = [label for label in pull_request_labels if label not in issue_backport_label]
+            filter_issue_labels = [label for label in issue_backport_label if label not in pull_request_labels]
+
+            if label_action == 'labeled':
+                for filter_pull_request_label in filter_pull_request_labels:
```

Comment:
instead of filtering and looking for the labels you can get the label that was 'labeled'/'unlabeled' under `github.event.label.name`